core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/EmbeddedDatabaseService.java

```diff
@@ -1,26 +1,48 @@
 package com.sequenceiq.cloudbreak.service.cluster;
 
+import java.util.Optional;
+
 import javax.inject.Inject;
 
+import org.apache.commons.lang3.ObjectUtils;
 import org.springframework.stereotype.Component;
 
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.database.DatabaseAvailabilityType;
 import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
 import com.sequenceiq.cloudbreak.domain.Template;
 import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceGroup;
+import com.sequenceiq.common.api.type.InstanceGroupType;
 
 @Component
 public class EmbeddedDatabaseService {
     @Inject
     private EntitlementService entitlementService;
 
     public EmbeddedDatabaseInfo getEmbeddedDatabaseInfo(String actorCrn, String accountId, Stack stack) {
-        if (entitlementService.embeddedDatabaseOnAttachedDiskEnabled(actorCrn, accountId)) {
-            Template template = stack.getPrimaryGatewayInstance().getInstanceGroup().getTemplate();
-            int volumeCount = template == null ? 0 : template.getVolumeTemplates().stream()
-                    .mapToInt(volume -> volume.getVolumeCount()).sum();
+        DatabaseAvailabilityType externalDatabase = ObjectUtils.defaultIfNull(stack.getExternalDatabaseCreationType(), DatabaseAvailabilityType.NONE);
+        if (DatabaseAvailabilityType.NONE == externalDatabase && entitlementService.embeddedDatabaseOnAttachedDiskEnabled(actorCrn, accountId)) {
+            int volumeCount = calculateVolumeCountOnGatewayGroup(stack);
             return new EmbeddedDatabaseInfo(volumeCount > 0, volumeCount);
         } else {
             return new EmbeddedDatabaseInfo(false, 0);
         }
     }
+
+    public EmbeddedDatabaseInfo getEmbeddedDatabaseInfo(Stack stack) {
+        if (stack.getCluster().getEmbeddedDatabaseOnAttachedDisk()) {
+            int volumeCount = calculateVolumeCountOnGatewayGroup(stack);
+            return new EmbeddedDatabaseInfo(volumeCount > 0, volumeCount);
```

Comment:
This could be simplified by passing the volume count only and calculate the true/false in the constructor. This would reduce the code duplication as well.